galley/pkg/runtime/processors/serviceentry/processor.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceentry
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/metadata"
+	"istio.io/istio/galley/pkg/runtime/processing"
+	"istio.io/istio/galley/pkg/runtime/processors/serviceentry/convert"
+	"istio.io/istio/galley/pkg/runtime/processors/serviceentry/locality"
+	"istio.io/istio/galley/pkg/runtime/processors/serviceentry/log"
+	"istio.io/istio/galley/pkg/runtime/processors/serviceentry/node"
+	"istio.io/istio/galley/pkg/runtime/processors/serviceentry/pod"
+	"istio.io/istio/galley/pkg/runtime/publish"
+	"istio.io/istio/galley/pkg/runtime/resource"
+	"istio.io/istio/galley/pkg/runtime/stats"
+	"istio.io/istio/pkg/mcp/snapshot"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+const (
+	// TODO(nmittler): What's the right collection to use? Should it be the same as the normal ServiceEntry collection?
+	collection = "istio/networking/v1alpha3/synthetic/serviceentries"
+	typeURL    = "type.googleapis.com/istio.networking.v1alpha3.ServiceEntry"
+)
```

Comment:
as discussed offline ... I'll remove the custom collection here in favor of just reusing the existing ServiceEntry collection. Pilot will know that these are special since: 1) it will have to create a client that listens on a different group (not "default") and 2) We'll add a special annotation to indicate that the ServiceEntry is synthetic.